src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostUriPresentationEndpointTest.cs

```diff
@@ -0,0 +1,269 @@
+                    namespace SomeProject;
+
+                    public class Component : Microsoft.AspNetCore.Components.ComponentBase
+                    {
+                    }
+                    """),
+                // The above will make the component exist, but the .razor file needs to exist too for Uri presentation
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"), """
+                    This doesn't matter
+                    """)
+            ],
+            uris: [new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"))],
+            expected: "<Component />");
+    }
+
+    [Fact]
+    public async Task Component_IntoCSharp_NoTag()
+    {
+        await VerifyUriPresentationAsync(
+            input: """
+                This is a Razor document.
+
+                <div>
+                </div>
+
+                @code {
+                    [||]
+                }
+                """,
+            additionalFiles: [
+                // The source generator isn't hooked up to our test project, so we have to manually "compile" the razor file
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.cs"), """
+                    namespace SomeProject;
+
+                    public class Component : Microsoft.AspNetCore.Components.ComponentBase
+                    {
+                    }
+                    """),
+                // The above will make the component exist, but the .razor file needs to exist too for Uri presentation
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"), """
+                    This doesn't matter
+                    """)
+            ],
+            uris: [new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"))],
+            expected: null);
+    }
+
+    [Fact]
+    public async Task Component_WithChildFile()
+    {
+        await VerifyUriPresentationAsync(
+            input: """
+                This is a Razor document.
+
+                <div>
+                    [||]
+                </div>
+
+                The end.
+                """,
+            additionalFiles: [
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.cs"), """
+                    namespace SomeProject;
+
+                    public class Component : Microsoft.AspNetCore.Components.ComponentBase
+                    {
+                    }
+                    """),
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"), """
+                    This doesn't matter
+                    """)
+            ],
+            uris: [
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor")),
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor.css")),
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor.js"))
```

Comment:
Consider another helper for all of these `new Uri(Path.Combine(...))` calls with `TestProject.SomeProject.Path`:

```C#
private static Uri FIlePathUri(string fileName)
    => new(Path.Combine(TestProjectData.SomeProjectPath, fileName));
```

Then they all clean up nice:

```suggestion
                FIlePathUri("Component.razor"),
                FIlePathUri("Component.razor.css"),
                FIlePathUri("Component.razor.js")
```

Reply:
I like your fancy ideas